Mesh4all/m4a-firmware
# examples: slipdev integration example

### Contribution description
This PR it's to add the slipdev example, to use the UART as net device.



### Testing procedure

when you are using pyterm console in your board, use:
ifconfig
```sh
> ifconfig
2022-01-13 14:30:46,017 # ifconfig
2022-01-13 14:30:46,020 # Iface  4 
2022-01-13 14:30:46,024 #           Long HWaddr: AE:4F:B5:37:F8:1D:6C:EF 
2022-01-13 14:30:46,027 #           MTU:65535  HL:64  
2022-01-13 14:30:46,029 #           Link type: wired
2022-01-13 14:30:46,035 #           inet6 addr: fe80::ac4f:b537:f81d:6cef  scope: link  VAL
2022-01-13 14:30:46,038 #           inet6 group: ff02::1
2022-01-13 14:30:46,041 #           inet6 group: ff02::1:ff1d:6cef
2022-01-13 14:30:46,042 #           
2022-01-13 14:30:46,045 #           Statistics for Layer 2
2022-01-13 14:30:46,048 #             RX packets 9  bytes 468
2022-01-13 14:30:46,052 #             TX packets 49 (Multicast: 0)  bytes 2580
2022-01-13 14:30:46,056 #             TX succeeded 0 errors 0
2022-01-13 14:30:46,058 #           Statistics for IPv6
2022-01-13 14:30:46,061 #             RX packets 9  bytes 468
2022-01-13 14:30:46,066 #             TX packets 49 (Multicast: 4)  bytes 2580
2022-01-13 14:30:46,069 #             TX succeeded 49 errors 0
2022-01-13 14:30:46,070 # 
```
then use ping to the link local address of the another board that is connected in the UART. in this example it's used fe80::e405:df5c:7fc4:2314 to the secondary board.
```sh
ping fe80::e405:df5c:7fc4:2314
```
expected outpu after pingt:
```sh
2022-01-13 14:30:59,292 # 12 bytes from fe80::e405:df5c:7fc4:2314%4: icmp_seq=0 ttl=64 time=10.384 ms
2022-01-13 14:31:00,291 # 12 bytes from fe80::e405:df5c:7fc4:2314%4: icmp_seq=1 ttl=64 time=9.962 ms
2022-01-13 14:31:01,291 # 12 bytes from fe80::e405:df5c:7fc4:2314%4: icmp_seq=2 ttl=64 time=9.970 ms
2022-01-13 14:31:01,292 # 
2022-01-13 14:31:01,296 # --- fe80::e405:df5c:7fc4:2314 PING statistics ---
2022-01-13 14:31:01,301 # 3 packets transmitted, 3 packets received, 0% packet loss

```



### Issues/PRs references
fixes #85
